templates/snmpd/snmpd.conf.erb

```diff
@@ -5,23 +5,23 @@
 ################################################################################
 # AGENT BEHAVIOUR
 
-<% if @master != nil -%>
+<% if @master.nil? -%>
```

Comment:
Is this change from checking if not nil to checking if nil correct?  Did you mean to check 
   unless @master.nil?

Numerous other of these types of check problems.  Would be caught by a thorough unit test